sabnzbd/nzbstuff.py

```diff
@@ -197,17 +197,18 @@ def get_article(self, server: Server, servers: List[Server]):
                     for server_check in servers:
                         if log:
                             logging.debug("Article %s | Server: %s | checking", self.article, server.host)
-                        if server_check.active and (server_check.priority < found_priority):
-                            if server_check.priority < server.priority:
-                                if not self.server_in_try_list(server_check):
-                                    if log:
-                                        logging.debug(
-                                            "Article %s | Server: %s | setting found priority to %s",
-                                            self.article,
-                                            server.host,
-                                            server_check.priority,
-                                        )
-                                    found_priority = server_check.priority
+                        if server_check.priority >= server.priority:
+                            break
+                        if not self.server_in_try_list(server_check):
+                            if log:
+                                logging.debug(
+                                    "Article %s | Server: %s | setting found priority to %s",
+                                    self.article,
+                                    server.host,
+                                    server_check.priority,
+                                )
+                            found_priority = server_check.priority
```

Comment:
Wouldn't be possible to maybe write unit tests? If we could do that we can test all kinds of situations.
Should be possible to construct list of Server's and see how the Article proceeds depending on their settings.

Reply:
Probably, but I have no knowledge of how to do it. Are there any existing tests that does a similar thing that I could look at?

Reply:
No, it would have to be made from scratch.